### PR TITLE
fix(sql): resolve SQL Workspace/AI Assistant duplicate OGC_FID Binder Error (#499)

### LINKED ISSUE #499: SQL Workspace and AI Assistant SQL Binder Error
I think these two are related to each other in the fact they have similar wording in their errors. Executing a query using the SQL Workspace with the DuckDB Engine can throw an unrecoverable error:
```
Binder Error: table "st_read" has duplicate column name "OGC_FID" LINE 1: ... REPLACE TEMP TABLE "sql_result_5_20_53_pm" AS SELECT * FROM ST_Read('__geolibre_sql_1781817661002_k0pkwht6ptr__sql_resu... ^
```

This is replicable across 3 machines and users. Steps to replicate:
1. Open SQL Workspace
2. Run a query against a dataset already loaded in the layer tab
3. Add that result as a layer
4. Run any other query. It fails with the above error.

The same general flow happens if you ask the AI assistant to create multiple new layers from an existing layer. It runs an SQL query, creates the first one, then all subsequent DuckDB queries it tries to run throws the `Binder Error`

Image

I haven't done any investigation into underlying causes beyond being able to replicate it.

### PR DESCRIPTION
**Pull request by giswqs**
Fixes #499.

## Problem

Running a query in the SQL Workspace (or having the AI assistant create layers from an existing layer), **adding the result as a layer**, then running **any** subsequent query failed with an unrecoverable error:

```
Binder Error: table "st_read" has duplicate column name "OGC_FID"
... CREATE OR REPLACE TEMP TABLE "sql_result_..." AS SELECT * FROM ST_Read('__geolibre_sql_..._sql_resu...
```

Reproducible across multiple machines/users.

## Root cause

The SQL Workspace exposes each loaded layer to DuckDB by writing its in-memory GeoJSON to the VFS and running `CREATE TEMP TABLE ... AS SELECT * FROM ST_Read(<file>)` (`registerLayerTables`).

GDAL's GeoJSON driver (which backs `ST_Read`) **synthesises an `OGC_FID` column** for every feature. When a query result is added back as a layer, that synthesised `OGC_FID` is carried into the new layer's GeoJSON **as a regular property**. On the next query, `registerLayerTables` re-reads that GeoJSON, GDAL emits its synthetic `OGC_FID` *again* alongside the existing property, and `ST_Read` produces a relation with two `OGC_FID` columns — which DuckDB rejects at bind time, so the table (and the whole query) fails.

## Fix

Strip the reserved GDAL FID column (`OGC_FID`) from each feature's properties before handing the collection to `ST_Read` in `registerLayerTables`. GDAL's synthetic FID then becomes the only `OGC_FID`, so re-reading round-tripped GeoJSON binds cleanly.

The helper (`stripAutoFidColumn`) lives in the wasm-free `duckdb-geometry` module so it is unit-testable under plain Node; it returns the input unchanged (no allocation) when no feature carries `OGC_FID`, and never mutates the caller's collection.

## Tests

- New `stripAutoFidColumn` unit tests in `tests/duckdb-geometry.test.ts` (strip, no-mutation, identity-when-absent, null-properties, multi-feature).
- Full frontend suite passes (1102 passing); `tsc -b` and pre-commit (incl. npm build) clean.

## Release Notes

* **Bug Fixes**
  * Improved geographic data handling by automatically removing auto-generated `OGC_FID` properties from GeoJSON features when loading and re-registering layers, keeping stored/exported layer data cleaner and avoiding related property duplication issues.
* **Tests**
  * Added unit tests to verify `OGC_FID` is stripped correctly, the original data isn’t mutated, empty collections are unchanged, and edge cases like missing properties are handled safely.

### CODE REVIEW

File: apps/geolibre-desktop/src/lib/sql-workspace.ts [hunk 1]
```diff
@@ -253,8 +254,13 @@ export async function registerLayerTables(
   const registered: SqlWorkspaceTable[] = [];
 
   for (const { layer, tableName } of assignTableNames(layers)) {
+    // assignTableNames only yields layers that carry a GeoJSON collection.
+    if (!layer.geojson) continue;
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/sql-workspace.ts]: Done in 8d5f538 — reworded the comment to state the guard's real purpose: it narrows the optional `layer.geojson` to `FeatureCollection` for the `stripAutoFidColumn` call, since TypeScript doesn't carry assignTableNames' filter narrowing across the function boundary.

File: apps/geolibre-desktop/src/lib/duckdb-geometry.ts [hunk 2]
```diff
@@ -110,3 +111,37 @@ export function geometryGeoJsonSql(
...
+  const features = geojson.features.map((feature) => {
+    const props = feature.properties;
+    if (props && GDAL_AUTO_FID_COLUMN in props) {
+      changed = true;
+      const { [GDAL_AUTO_FID_COLUMN]: _omit, ...rest } = props;
+      return { ...feature, properties: rest };
+    }
+    return feature;
+  });
+  return changed ? { ...geojson, features } : geojson;
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/duckdb-geometry.ts]: Done in 8d5f538 — switched to a `.some()` pre-scan with an early `return geojson`, so the no-OGC_FID path no longer allocates the throw-away `.map()` array, matching the documented zero-allocation promise.

File: tests/duckdb-geometry.test.ts [hunk 3]
```diff
@@ -129,3 +131,70 @@ describe("geometryGeoJsonSql", () => {
...
+
+  it("returns the same object when no feature carries OGC_FID", () => {
+    const input = collection({ name: "a" });
+    assert.equal(stripAutoFidColumn(input), input);
+  });
+
+  it("tolerates features with null properties", () => {
+    const out = stripAutoFidColumn(collection(null));
+    assert.equal(out.features[0].properties, null);
+  });
```
[the remark this answers is not in the document]
Comment (giswqs) [reply tests/duckdb-geometry.test.ts:171]: Done in 8d5f538 — added the empty feature-collection test asserting the original object is returned.

File: apps/geolibre-desktop/src/lib/duckdb-geometry.ts [hunk 4]
```diff
@@ -110,3 +111,42 @@ export function geometryGeoJsonSql(
...
+  const features = geojson.features.map((feature) => {
+    const props = feature.properties;
+    if (props && GDAL_AUTO_FID_COLUMN in props) {
+      const { [GDAL_AUTO_FID_COLUMN]: _omit, ...rest } = props;
+      return { ...feature, properties: rest };
+    }
+    return feature;
+  });
+  return { ...geojson, features };
+}
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/duckdb-geometry.ts:152]: Done in 9f03a52 — added `key === GDAL_AUTO_FID_COLUMN` to the skip-list in `rowsToFeatureCollection` so a SQL-result layer never acquires the OGC_FID artefact (no more attribute-table/export leak, and it can't re-trigger the Binder Error). Kept `stripAutoFidColumn` at registration time as the belt-and-braces guard for layers ingested via other paths (drag-drop, project restore, plugins) that already carry OGC_FID, exactly as you suggested.

File: tests/duckdb-geometry.test.ts [hunk 5]
```diff
@@ -129,3 +131,75 @@ describe("geometryGeoJsonSql", () => {
...
+        },
+      ],
+    };
+    const out = stripAutoFidColumn(input);
+    assert.deepEqual(
+      out.features.map((f) => f.properties),
+      [{ name: "a" }, { name: "b" }, { name: "c" }],
+    );
+  });
+});
```
[the remark this answers is not in the document]
Comment (giswqs) [reply tests/duckdb-geometry.test.ts:213]: Done in 9f03a52 — added a test asserting empty-object properties when OGC_FID is the sole property, and a non-mutation assertion (clone + deepEqual) inside the multi-feature test.
